src/main/java/org/jabref/logic/importer/fetcher/IEEE.java

```diff
@@ -234,8 +228,31 @@ public String getName() {
 
     @Override
     public URL getComplexQueryURL(ComplexSearchQuery complexSearchQuery) throws URISyntaxException, MalformedURLException {
+        return getComplexQueryURL(complexSearchQuery, 0);
```

Comment:
This should be the default implementation, i.e. push to `PagedSearchBasedParserFetcher`

Reply:
> You have currently quite a lot of code duplications. The main sources are:
> 
> 1. Complex and string-based queries have to be implemented separately, leading to mostly the same code. Suggestion: remove string-based queries completely
> 2. Unpaged search methods always fall back to paged search method with page number 0. Suggestion: put these fallbacks in the general `PagedSearchFetcher` interface.

Thanks for your feedback! :)
Regarding 1: 
I think that this is a good idea but will require more work than just replacing the method. 
This is because for the WebSearchPane for instance the normal string-based search is used as a fallback case when the query could not be parsed.
Therefore I do not feel comfortable just tossing the string-based version.
However, I will address this in the upcoming weeks :)!

Reply:
I implemented all other suggestions

Reply:
Good!

Concerning the first point, doesn't it work to parse the queries as follows:
`author=me and title=something` -> `ComplexQuery[author: "me", title: "something", rest: ""]`
`author=me something` -> `ComplexQuery[author: "me", title: "", rest: "something"]`
`something` -> `ComplexQuery[author: "", title: "", rest: ""something]`

Then you don't need any fall-back to a purely string-based search.

Reply:
I now removed the normal performSearch all together :)

Reply:
And tried to test it throughout.

Reply:
Thanks a lot! If you now also remove the "complex" in the names, I'm super happy ;-)

Reply:
I now removed the complex from perform search :)
Regarding the query object, I believe that it is sensible to keep that name as it will be extended in the future such as structured information.